python/lsst/dax/apdb/apdbSqlSchema.py

```diff
@@ -94,9 +94,14 @@ def process_bind_param(self, value: Any, dialect: sqlalchemy.Dialect) -> Optiona
         else:
             return "%.32x" % value.int
 
-    def process_result_value(self, value: Optional[str], dialect: sqlalchemy.Dialect) -> Optional[uuid.UUID]:
+    def process_result_value(
+        self, value: str | uuid.UUID | None, dialect: sqlalchemy.engine.Dialect
+    ) -> Optional[uuid.UUID]:
         if value is None:
             return value
+        elif isinstance(value, uuid.UUID):
+            # sqlalchemy 2 converts to UUID internally
```

Comment:
Same question as on daf_butler: this says SA2 but presumably still works fine in 1.4.